src/components/PopupConfirm.js

```diff
@@ -6,16 +6,23 @@ export default class PopupConfirm extends Popup {
     this._confirmDeleteButton = this._modal.querySelector(
       ".modal-confirm__submit-button"
     );
+    this._submitButton = this._modal.querySelector(".modal__submit-button");
   }
 
   handleConfirm(action) {
     this._handleConfirmDelete = action;
   }
 
+  handleButtonLoading() {
+    this._submitButton.textContent = "Saving...";
+    setTimeout(() => {
```

Comment:
Please, delete `setTimeout` because you need to really wait till the server request is fullfilled

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.